spec/controllers/vm_infra_controller_spec.rb

```diff
@@ -328,6 +328,43 @@
     expect(response.status).to eq(200)
   end
 
+  context 'transform VM dialog' do
+    let(:dialog)          { double("Dialog") }
+    let(:resource_action) { double("ResourceAction") }
+
+    before do
+      allow(Dialog).to receive(:find_by).and_return(dialog)
+      allow(dialog).to receive(:id).and_return(770)
+      allow(dialog).to receive(:resource_actions).and_return([resource_action])
+      allow(dialog).to receive(:target_resource=)
+      allow(dialog).to receive(:init_fields_with_values)
+      allow(dialog).to receive(:dialog_tabs).and_return([])
+      allow(dialog).to receive(:buttons).and_return('')
+      allow(resource_action).to receive(:id).and_return(42)
+      allow(resource_action).to receive(:dialog).and_return(dialog)
+      allow(resource_action).to receive(:dialog_id).and_return(dialog.id)
+    end
```

Comment:
You don't need these mocks if you use factories.

Reply:
@skateman Indeed, a factory for `ResourceAction` already exists: https://github.com/ManageIQ/manageiq/blob/master/spec/factories/resource_action.rb